kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/KinClient.java

```diff
@@ -111,10 +108,26 @@ private void loadAccounts() {
             e.printStackTrace();
         }
         if (accounts != null && !accounts.isEmpty()) {
-            for (KeyPair account : accounts) {
-                kinAccounts.add(createNewKinAccount(account));
+            updateKinAccounts(accounts);
+        }
+    }
+
+    private void updateKinAccounts(List<KeyPair> storageAccounts) {
+        Map<String, KinAccountImpl> accountsMap = new HashMap<>();
```

Comment:
BTW, you should be able to do this just with `Set`, but I can see that we have only `equals` implemented without `hashCode`, so probably that's why you didn't use it.

Reply:
yes, although we can generate a hashCode with the help of the IDE.
Keep it that way or to change?

Reply:
not critical at all, your call